components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth2/dao/SQLQueries.java

```diff
@@ -164,7 +164,8 @@
             "TOKEN_STATE_ID=? WHERE TOKEN_ID=?";
 
     public static final String REVOKE_ACCESS_TOKEN = "UPDATE IDN_OAUTH2_ACCESS_TOKEN SET TOKEN_STATE=?, " +
-                                                     "TOKEN_STATE_ID=? WHERE ACCESS_TOKEN=?";
+                                                     "TOKEN_STATE_ID=? WHERE CONSUMER_KEY_ID = (SELECT ID FROM " +
+            "IDN_OAUTH_CONSUMER_APPS WHERE CONSUMER_KEY = ?) AND TOKEN_STATE=?";
```

Comment:
We can't change this query. It is used to revoke individual access tokens. Since you need to revoke all the access tokens issued for a application add a seperate query.